src/sparsezoo/analysis/analysis.py

```diff
@@ -0,0 +1,219 @@
+    zero_point: int = Field(
+        description="Node zero point for quantization, default zero"
+    )
+    dtype: Optional[str] = Field(
+        description="This node's parameter's data type", default=None
+    )
+
+
+class ModelAnalysis(BaseModel):
+    """
+    Pydantic model for analysis of a model
+    """
+
+    layer_counts: Dict[str, int] = Field(description="Overview of layers", default={})
+    operation_counts: Dict[str, int] = Field(
+        description="Overview of operations (nodes that are not layers)", default={}
+    )
+    num_ops: int = Field(
+        description="The total number of (floating point and/or integer) operations "
+        "performed in one forward pass of the model"
+    )
+    average_sparsity: float = Field(
+        description="Average sparsity across all trainable parameters "
+        "(excluding biases)"
+    )
+    average_four_block_sparsity: float = Field(
+        description="Average sparsity four block sparsity across all "
+        "trainable parameters (excluding biases)"
+    )
+    num_sparse_layers: int = Field(description="Number of layers with any sparsity")
+    num_quantized_layers: int = Field(description="Number of quantized layers")
+    num_parameters: int = Field(
+        description="Total number of all parameters (excluding biases)"
+    )
+    num_sparse_parameters: int = Field(
+        description="Total number of all sparsified parameters (excluding biases)"
+    )
+    num_four_blocks: int = Field(
+        description="Total number of all four blocks (excluding biases)"
+    )
+    num_sparse_four_blocks: int = Field(
+        description="Total number of all sparsified four blocks (excluding biases)"
+    )
+
+    layers: List[NodeAnalysis] = Field(
+        description="List of analyses for each layer in the model graph", default=[]
+    )
+
+    @classmethod
+    def from_onnx_model(cls, onnx_file_path: str):
+        """
+        Class constructor for Model Analysis
+
+        :param cls: class being constructed
+        :param onnx_file_path: path to onnx file being analyzed
+        :return: instance of ModelAnalysis class
+        """
+        model_onnx = onnx.load(onnx_file_path)
+
+        param_prunable_nodes = [
+            node
+            for node in model_onnx.graph.node
+            if is_parameterized_prunable_layer(model_onnx, node)
+        ]
+        sparsity_sizes = numpy.array(
+            [
+                get_node_num_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+        four_block_sparsity_sizes = numpy.array(
+            [
+                get_node_num_four_block_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+
+        num_sparse_parameters, num_parameters = numpy.sum(sparsity_sizes, axis=0)
+        num_sparse_four_blocks, num_four_blocks = numpy.sum(
+            four_block_sparsity_sizes, axis=0
```

Comment:
can this section be completed by using the `NodeAnalysis` list rather than re-doing here?